### DISCUSSION TITLE
Muttiple result in startnotification

### DISCUSSION THREAD

**Discussion by dmsuh** [capacitor-community/bluetooth-le#517] [Q&A] [answered]
During the work of startstartNotifications, callback is called many times, how to combine data?
![image](https://user-images.githubusercontent.com/90576894/230419956-fbe1396c-8bc6-4a39-9e22-d69042444fc2.png)
And it should have been, as far as I understand, so
![image](https://user-images.githubusercontent.com/90576894/230418258-432a82ca-35b1-4879-b863-a0dc8d65805b.png)
  code 
```bash
const al = await BleClient.startNotifications(
      id,
      numberToUUID(0xffe0), numberToUUID(0xffe1),
      (value) => {
        //data.value = data.value.concat(value.buffer[Symbol.toStringTag])
        console.log("Discrpition :",value.buffer)
      }
    )
 ```

**Comment by pwespi** [comment]
> how to combine data

That depends on what you want to do with the data. If you want the data as numbers, you can use the `dataViewToNumbers` helper method to convert from `DataView` to `number[]` and then use e.g. `concat` or `flat` method on the arrays.

e.g.
```ts
// values: DataView[]
const combined = values.map(dataViewToNumbers).flat();
```

**Comment by dmsuh** [comment]
I need to get the information in the DataView in order to parse it later, the size of the DataView changes, so it would be desirable for me to get a full block at once and not 20 bytes at a time, but a full one at once. To then use getUnit32(), getFloat16.... parse the data from the device. And how to combine the multiple output of this DataView I do not understand. I need one final DataView, since it's hard to iterate with data, especially if everything comes from the device, even some parts that don't need to be read

**Comment by dmsuh** [comment]
i need one value that constain {AB 02 AC 7D F9 5E 42 DA E6 42 42 B7 11 02 80 31 03 5F 42 26 FA 42 42 00 00 CE 0A 5D 00 00 06 0F 8A 14 F9 5E 42 D2 E6 42 42 00 80 0C 07 5D 00 00 04 09 ED}

**Comment by dmsuh** [comment, marked as answer]
I made this decision, if you suggest something better then thank you
```bash
const al= await BleClient.startNotifications(
      id,
      numberToUUID(0xffe0), numberToUUID(0xffe1),
      (value:DataView) => {
        const flag=value.byteLength-1;
        data.value.push(value)
        if(value.getUint8(flag)===237){
          const tmp=data.value.map(dataViewToHexString).flat(Infinity);
          let tmpStr='';
          for (let i = 0; i < tmp.length; i++) {
              tmpStr+=tmp[i]
              tmpStr+=' '
          }
          const dtNew:DataView=hexStringToDataView(tmpStr)
          dataResolver(dtNew);
          data.value=[];
        }
      }
    )
```

**Comment by dmsuh** [comment]
237 end data from device
